boltons/dictutils.py

```diff
@@ -964,6 +964,49 @@ def subdict(d, keep=None, drop=None):
 
     keys = set(keep) - set(drop)
 
-    return dict([(k, v) for k, v in d.items() if k in keys])
+    return type(d)([(k, v) for k, v in d.items() if k in keys])
+
+
+class FrozenDict(dict):
```

Comment:
https://www.python.org/dev/peps/pep-0416/ linking to the rejected PEP from a docstring would probably be helpful for discoverability

it also links to a BUNCH of implementations to the point that it even categorizes them into whitelist/blacklist

the PEP points out one issue with the inherit-from-dict is that you can use `dict.__setitem__(frozen, key, val)` to "sneak around" the immutability

I'm not sure if that is really an issue, or maybe something worth documenting

Reply:
Haha of course there'd be a PEP about this. Not sure looking at it would have sped things up though, probably would have just read a whole bunch of conflicting approaches instead of taking my own :P

Also, this is Python and there's lots of ways to sneak around. No point in chasing them all down. But also why document it when it's probably just going to enable it? :P